src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -181,6 +182,92 @@ public UserDefinedFunction WithBinding(INameResolver nameResolver, IBinderGlue b
             return func;
         }
 
+        private static readonly ISet<string> _restrictedUDFNames = new HashSet<string> { "Type", "IsType", "AsType" };
+
+        /// <summary>
+        /// Helper to create IR UserDefinedFunctions.
+        /// </summary>
+        /// <param name="uDFs">Valid Parsed UDFs to be converted into UserDefinedFunction.</param>
+        /// <param name="errors">Errors when creating functions.</param>
+        /// <returns>IEnumerable of UserDefinedFunction.</returns>
+        public static IEnumerable<UserDefinedFunction> CreateFunctions(IEnumerable<UDF> uDFs, out List<TexlError> errors)
+        {
+            Contracts.AssertValue(uDFs);
+            Contracts.AssertAllValues(uDFs);
+
+            var userDefinedFunctions = new List<UserDefinedFunction>();
+            var texlFunctionSet = new TexlFunctionSet();
+            errors = new List<TexlError>();
+
+            foreach (var udf in uDFs)
+            {
+                Contracts.Assert(udf.IsParseValid);
+
+                var udfName = udf.Ident.Name;
+                if (_restrictedUDFNames.Contains(udfName) || texlFunctionSet.AnyWithName(udfName) || BuiltinFunctionsCore._library.AnyWithName(udfName) || BuiltinFunctionsCore.OtherKnownFunctions.Contains(udfName))
+                {
+                    errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionAlreadyDefined, udfName));
+                    continue;
+                }
```

Comment:
Would this prevent us from introducing new functions in the future? For example, if someone has a UDF today called 'Summarize' in an app today, will it break when we introduce a function with that name?

Reply:
yes you are right .

Reply:
In general, the solution to this is that we always bind to UDF first before built-in functions.  I think there is a issue tracking this somewhere already.  But yes this check needs to be removed.